ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/ServerVersionUtil.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.tools.migrations.util;
+
+import io.confluent.ksql.api.client.Client;
+import io.confluent.ksql.api.client.ServerInfo;
+import io.confluent.ksql.tools.migrations.MigrationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ServerVersionUtil {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ServerVersionUtil.class);
+
+  private ServerVersionUtil() {
+  }
+
+  public static ServerInfo getServerInfo(final String ksqlServerUrl) {
+    final Client client = MigrationsUtil.getKsqlClient(ksqlServerUrl);
+    try {
+      return getServerInfo(client, ksqlServerUrl);
+    } finally {
+      client.close();
+    }
+  }
+
+  public static ServerInfo getServerInfo(final Client ksqlClient, final String ksqlServerUrl) {
+    final CompletableFuture<ServerInfo> response = ksqlClient.serverInfo();
+
+    try {
+      return response.get();
+    } catch (InterruptedException e) {
+      throw new MigrationException("Interrupted while attempting to connect to "
+          + ksqlServerUrl + "/info");
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof IllegalStateException) {
+        throw new MigrationException(e.getCause().getMessage()
+            + "\nPlease ensure that " + ksqlServerUrl + " is an active ksqlDB server and that the "
+            + "version of the migration tool is compatible with the version of the ksqlDB server.");
+      }
+      throw new MigrationException("Failed to query " + ksqlServerUrl + "/info: " + e.getMessage());
+    }
+  }
+
+  public static boolean isSupportedVersion(final String ksqlServerVersion) {
+    final KsqlServerVersion version;
+    try {
+      version = new KsqlServerVersion(ksqlServerVersion);
+    } catch (IllegalArgumentException e) {
+      throw new MigrationException("Could not parse ksqlDB server version to "
+          + "verify compatibility. Version: " + ksqlServerVersion);
+    }
+
+    return version.isAtLeast(6, 0, 0, 10);
+  }
+
+  public static boolean versionSupportsMultiKeyPullQuery(final String ksqlServerVersion) {
```

Comment:
This method will be used by `migrations info` in order to determine whether to issue a single, multi-key pull query to fetch version info, or whether we'll need to fall back to multiple, single-key lookups instead.